aws/rust-runtime/aws-config/src/provider_config.rs

```diff
@@ -252,11 +263,25 @@ impl ProviderConfig {
 
     /// Override the HTTPS connector for this configuration
     ///
+    /// **Warning**: Use of this method will prevent you from taking advantage of the timeout machinery.
+    /// Consider `with_tcp_connector`.
+    ///
     /// ## Note: Stability
     /// This method is expected to change to support HTTP configuration
-    pub fn with_connector(self, connector: DynConnector) -> Self {
+    pub fn with_http_connector(self, connector: DynConnector) -> Self {
+        ProviderConfig {
+            connector: HttpConnector::Prebuilt(Some(connector)),
+            ..self
+        }
+    }
+
+    /// Override the TCP connector for this configuration
+    ///
+    /// ## Note: Stability
```

Comment:
To be consistent with the standard library's docs:
```suggestion
    /// # Stability
```